pkgs/applications/science/chemistry/xcrysden/default.nix

```diff
@@ -0,0 +1,66 @@
+{ lib
+, fetchurl
+, fftw
+, gcc8Stdenv
+, gfortran
+, libGL
+, libGLU
+, tcl
+, tk
+, togl
+, util-linux
+, xorg
+}:
+
+let
+  bwidget = rec {
+    ver = "1.9.13";
+    tgz = "bwidget-${ver}.tar.gz";
+    src = fetchurl {
+      url = "http://sourceforge.net/projects/tcllib/files/BWidget/${ver}/${tgz}";
```

Comment:
I think there is a sourceforge mirror that people normally use.  If you grep through Nixpkgs for `mirror://sourceforge`, you should hopefully find some examples of how to write this.

Reply:
Done at https://github.com/NixOS/nixpkgs/pull/218239/commits/66a91f278a97e4fb991d1411a6066fca40a5d354#diff-f59da3f9121929af98b7e2d8fd1b933b7e2a8fdbd2c6f2b5bcb338685f4d292eR15
(rebased).